docs/parser_plugins.md

```diff
@@ -1 +1,210 @@
+- the **entry point** (instance of `EntryPoint`) is responsible for _registering_ the target functionality. It therefore also retains metadata like its name, a description and most importantly, the file matching directives. It is typically located in the `__init__.py` of the relevant functionality folder (see folder structure).
+    - the **entry point group** bundles several entry points together. By default, NOMAD scans all plugins under the group name `project.entry-points.'nomad.plugin'`.
+- the **module setup file** _exposes_ the entry point (and its group) under the format of `<module_name>.<functionality_folder_name>:<entry_point_name>`. This is the name by which you should call this entry point. In NOMAD we use the `pyproject.toml` setup file under the module's root folder.
+- the **NOMAD configuration file** called `nomad.yaml` allows further control over which entry points to _include_ or _exclude_, as well as their _configuration parameters_.
+
+Given a typical folder lay-out for a parser project, the players are located in
+...
+
+note on file matching
+
+For more detail on the specifics of each step, check out our [documentation]().
+
+## From Hierarchies to Schema
+
+As parsing involves the mapping between two well-defined formats, one could expect it to be just that trivial.
+In practice, however, parser developers have to manage discrepancies in semantics, shape, data type or units.
+This has lead to five distinct categories of responsibility for the developer to manage:
+
+1. select the relevant files and read them into Python.
+2. target and extract relevant data fields within the source file format on a per-file basis.
+3. match the data of interest with their counterparts in the target/NOMAD schema.
+4. reshape and mangle the data to match the target/NOMAD `Quantity`s' specification. This may include computing derived properties not present in the original source files.
+5. build up a Python `archive` object using the classes provided by the target/NOMAD schema.
+
+In the past, we encountered cases where the delineation between these responsibilities blurred, resulting in widely different parser designs.
+Especially when dealing with the larger, more feature-rich parsers, these designs could become quite complex.
+
+During refactoring, we therefore addressed these responsibilities on a more individual basis.
+This lead to a clear defined _interfaces_ and powerful _tools_ for the developer.
+These are further complimented by _best practices_ and _design protocols_.
+
+More specifically, point 1 (file selection) is now handled by the `MatchingParser` class, from which the parser class inherits. It coordinates with the NOMAD base to scan the uploaded folder tree and select the relevant files.
```

Comment:
I'm afraid that it is a bit difficult overwhelming to take in all these details without even having started any example. I think we need to have a bit more of a (extended) how-to framework, then the MatchingParser class will naturally come up as you set up the mock parser. Is this possible?

Reply:
The problems tackled and their solutions can be separated, yes.
I'll prep some code snippet + introduce a subsection for the solutiuons. Bernadette's figure may also help here.

Reply:
Turned the solutions into a list of their own. Will now add a short example showing solutions 1, 2, and 5 working together.